### PR TITLE
Fix paths to config templates

### PR DESCRIPTION
**Pull request by mnagy**
@praiskup @bparees fyi

### CODE REVIEW

**Comment by praiskup** [comment]
Yeah, one issue I see here that we generate configuration file which
is not by default on persistent storage.  Would this be possible to move into
/var/lib/pgsql/data/userdata?

**Comment by praiskup** [comment]
- s/$CONTAINER_SCRIPTS_PATH/$PGDATA/
- fix envsubst

**Comment by mnagy** [comment]
We do not want that. We want the configuration file to be regenerated every time the container is started, putting it in persistent volume doesn't make much sense. That takes care of changing environment variable changes, image updates and so on.

**Comment by mfojtik** [comment]
@mnagy we should allow volume with user specified configuration file and in that case, our entrypoint will just skip all env variables and leave the user with his pain :-)

**Comment by mnagy** [comment]
@mfojtik that sounds reasonable. Not in this PR though :)

**Comment by praiskup** [comment]
I still don't see a reason why we have to have another fix-permissions'ed variable
user potentially have to care about?

**Comment by mnagy** [comment]
@praiskup I don't understand. What do you mean by fix-permisions'ed variable?

**Comment by praiskup** [comment]
s/variable/directory/

**Comment by mnagy** [comment]
I still don't follow. Can you elaborate?

**Comment by praiskup** [comment]
To pass the testsuite, you need to allow group 0 to write into
/usr/share/container-scripts/postgresql
-- so (a) /var/lib/pgsql, (b) /var/run/postgresql and now (c).

**Comment by mnagy** [comment]
PTAL

**Comment by praiskup** [comment]
Looks fine.

**Comment by mfojtik** [comment]
@mnagy is there any test case for this?

**Comment by mnagy** [comment]
@mfojtik I've tested it manually. There is a PR for extended tests here: https://github.com/openshift/origin/pull/5874

File: 9.2/root/usr/share/container-scripts/postgresql/common.sh [hunk 1]
```diff
@@ -80,6 +80,12 @@ function generate_postgresql_config() {
       > "${POSTGRESQL_CONFIG_FILE}"
 }
 
+function generate_postgresql_recovery_config() {
+  envsubst \
+      < "${CONTAINER_SCRIPTS_PATH}/openshift-custom-recovery.conf.template" \
```
Comment (bparees) [inline 9.2/root/usr/share/container-scripts/postgresql/common.sh]: i don't see this file, where is it coming from?

**Pull request merged**